### PR TITLE
Modify the construction of WebDriver and WebDriverWait so that we are…

### PR DESCRIPTION
**Pull request by mikeSmith08**
… wrapping it and can then control its creation and closing for BrowserStack tests.

This is all in preparation for being able to run the Compatibility Tests directly via a test launcher in Java so that we can properly multi-thread the test runs and query the results ourselves to be able to figure out if we need to run and individual test or not. Whilst I haven't committed it in this PR, I can confirm that using the application context and local/browserstack variants like this, we can get multiple compatibility tests running.

### CODE REVIEW

File: src/test/java/uk/gov/dhsc/htbhf/steps/Hooks.java [hunk 1]
```diff
@@ -1,16 +1,25 @@
 package uk.gov.dhsc.htbhf.steps;
 
 import io.cucumber.java.After;
+import io.cucumber.java.Before;
 
 public class Hooks extends BaseSteps {
+
+    @Before
+    public void initDriver() {
+        if ("browserstack".equals(activeProfile)) {
```
Comment (YetAnotherMatt) [inline src/test/java/uk/gov/dhsc/htbhf/steps/Hooks.java]: Suggest pulling the check into a method

Comment (mikeSmith08) [reply src/test/java/uk/gov/dhsc/htbhf/steps/Hooks.java]: Yep, I'll do it in the next PR so I don't dismiss any reviews.

File: src/test/java/uk/gov/dhsc/htbhf/BrowserStackDriverWrapper.java [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+    private final URL browserStackUrl;
+
+    private WebDriver webDriver;
+
+    private WebDriverWait webDriverWait;
+
+    public BrowserStackDriverWrapper(String browserStackUser, String browserStackKey, int waitTimeoutInSeconds) {
+        Validate.notBlank(browserStackUser, "BrowserStack user must be provided, set the BROWSER_STACK_USER environment variable");
+        Validate.notBlank(browserStackKey, "BrowserStack key must be provided, set the BROWSER_STACK_KEY environment variable");
+        log.info("Using login: {} {}", browserStackUser, browserStackKey);
```
Comment (dwybourn) [inline src/test/java/uk/gov/dhsc/htbhf/BrowserStackDriverWrapper.java]: Should be logging credentials.

Comment (mikeSmith08) [reply src/test/java/uk/gov/dhsc/htbhf/BrowserStackDriverWrapper.java]: Yes, I'll remove in the next PR where I've fixed this and added logback-classic to the classpath so the logging actually works. The implication there is that this doesn't get logged at the moment so I'm happy to wait for the next PR.

**Pull request merged**